src/ucs/memory/memtype_cache.c

```diff
@@ -300,6 +315,30 @@ UCS_PROFILE_FUNC(ucs_status_t, ucs_memtype_cache_lookup,
     return status;
 }
 
+int ucs_memtype_cache_is_empty()
+{
+    return (ucs_memtype_cache_global_instance &&
+            !ucs_memtype_cache_global_instance->pgtable.num_regions);
+}
+
+static ucs_status_t ucs_memtype_cache_init_handler()
+{
+    ucs_status_t status;
+
+    status = ucm_set_event_handler(UCM_EVENT_MEM_TYPE_ALLOC |
+                                   UCM_EVENT_MEM_TYPE_FREE |
+                                   UCM_EVENT_FLAG_EXISTING_ALLOC,
+                                   1000, ucs_memtype_cache_event_callback,
+                                   ucs_memtype_cache_global_instance);
+    if ((status != UCS_OK) && (status != UCS_ERR_UNSUPPORTED)) {
+        ucs_error("failed to set UCM memtype event handler: %s",
+                  ucs_status_string(status));
+        ucs_pgtable_cleanup(&ucs_memtype_cache_global_instance->pgtable);
```

Comment:
this function did not init page table, so it should not clean it
need to move it outside this function

Reply:
removed this too